contracts/managers/price-triggers/RSIMidlineCrossTrigger.sol

```diff
@@ -0,0 +1,104 @@
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { IPriceTrigger } from "./IPriceTrigger.sol";
+import { IOracle } from "../../meta-oracles/interfaces/IOracle.sol";
+import { IMetaOracleV2 } from "../../meta-oracles/interfaces/IMetaOracleV2.sol";
+
+
+/**
+ * @title RSIMidlineCrossTrigger
+ * @author Set Protocol
+ *
+ * Implementing the IPriceTrigger interface, this contract is queried by a
+ * RebalancingSetToken Manager to determine the amount of base asset to be
+ * allocated to by checking if the the RSI is above or below certain values.
+ *
+ * Below the RSI level of lowerBound means the RebalancingSetToken should be in the
+ * quote asset, above the upperBound the RebalancingSetToken should be
+ * in the base asset.
+ */
+contract RSIMidlineCrossTrigger is
+    IPriceTrigger
+{
+    /* ============ Constants ============ */
+    uint256 constant MAX_BASE_ASSET_ALLOCATION = 100;
+    uint256 constant MIN_BASE_ASSET_ALLOCATION = 0;
+
+    /* ============ State Variables ============ */
+    IMetaOracleV2 public rsiOracleInstance;
+    uint256 public lowerBound;
+    uint256 public upperBound;
+    uint256 public rsiTimePeriod;
+
+    /*
+     * RSIMidlineCrossTrigger constructor.
+     *
+     * @param  _rsiOracleInstance        The address of RSI oracle
+     * @param  _lowerBound               Lower bound of RSI to trigger a rebalance
+     * @param  _upperBound               Upper bound of RSI to trigger a rebalance
+     * @param  _rsiTimePeriod            The amount of days to use in RSI calculation
+     */
+    constructor(
+        IMetaOracleV2 _rsiOracleInstance,
+        uint256 _lowerBound,
+        uint256 _upperBound,
+        uint256 _rsiTimePeriod
+    )
+        public
+    {
+        // Check that upper bound value must be greater than lower bound value
+        require(
+            _upperBound > _lowerBound,
+            "RSIMidlineCrossTrigger.constructor: Upper bound must be greater than lower bound"
+        );
+
+        // Set all state variables
+        rsiOracleInstance = _rsiOracleInstance;
+        lowerBound = _lowerBound;
+        upperBound = _upperBound;
+        rsiTimePeriod = _rsiTimePeriod;
+    }
+
+    /*
+     * Returns the percentage of base asset the calling Manager should allocate the RebalancingSetToken
+     * to. If RSI is above upper bound then should be 100% allocated to base asset, if
+     * RSI is below lower bound then should be 0% allocated to base asset. Else function reverts.
+     *
+     * @return             The percentage of base asset to be allocated to
+     */
+    function checkPriceTrigger()
+        external
+        view
+        returns (uint256)
+    {
+        // Query RSI oracle
+        uint256 rsiValue = rsiOracleInstance.read(rsiTimePeriod);
+
+        // Check RSI value is above upper bound or below lower bound to trigger a rebalance
+        require(
+            rsiValue >= upperBound || rsiValue <= lowerBound,
```

Comment:
The lower bound should just be `rsiValue < lowerBound` because due to Solidity rounding 40.8 would register as a valid signal when it shouldn't